services/src/atdd-staging/src/test/java/org/openkilda/atdd/staging/tests/flow_crud/FlowCrudOverActiveSwitchesTest.java

```diff
@@ -0,0 +1,114 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package org.openkilda.atdd.staging.tests.flow_crud;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import cucumber.api.CucumberOptions;
+import cucumber.api.java.After;
+import cucumber.api.java.Before;
+import org.junit.runner.RunWith;
+import org.openkilda.atdd.staging.cucumber.CucumberWithSpringProfile;
+import org.openkilda.atdd.staging.model.topology.TopologyDefinition;
+import org.openkilda.atdd.staging.service.traffexam.OperationalException;
+import org.openkilda.atdd.staging.tests.AbstractFlowBasedHook;
+import org.openkilda.messaging.info.event.IslChangeType;
+import org.openkilda.messaging.info.event.IslInfoData;
+import org.openkilda.messaging.info.event.PathNode;
+import org.openkilda.messaging.info.event.SwitchInfoData;
+import org.openkilda.messaging.info.event.SwitchState;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Stream;
+
+
+@RunWith(CucumberWithSpringProfile.class)
+@CucumberOptions(features = {"classpath:features/flow_crud.feature"},
+        glue = {"org.openkilda.atdd.staging.tests.flow_crud", "org.openkilda.atdd.staging.steps"})
+@ActiveProfiles("mock")
+public class FlowCrudOverActiveSwitchesTest {
+
+    public static class FlowCrudOverActiveSwitchesHook extends AbstractFlowBasedHook {
+
+        @Before
+        public void prepareMocks() throws IOException {
+            setup5SwitchTopology();
+
+            mockFlowInTE("sw1", 10, "sw2", 10, 1);
+            mockFlowInTE("sw1", 10, "sw3", 10, 2);
+
+            mockFlowCrudInNorthbound();
+
+            mockTraffExam();
+
+            removedFlows.clear();
+        }
+
+        private void setup5SwitchTopology() throws IOException {
```

Comment:
Is this yet another way to setup a topology?  How many ways in the test libraries do we have of setting up a topology?